judgments/tests/test_detail.py

```diff
@@ -131,6 +131,53 @@ def test_pdf_link_with_no_size(self, mock_judgment, mock_pdf_size):
         self.assertIn("/test/2023/123/data.pdf", decoded_response)
 
 
+@pytest.mark.django_db
+class TestDocumentDownloadOptions:
+    @patch("judgments.views.detail.get_pdf_uri")
+    @patch("judgments.views.detail.get_pdf_size")
+    @patch("judgments.views.detail.get_judgment_by_uri")
+    @pytest.mark.parametrize(
+        "uri,document_type",
+        [("eat/2023/1/press-summary/1", "press summary"), ("eat/2023/1", "judgment")],
+    )
+    def test_download_options(
+        self, mock_judgment, mock_get_pdf_size, mock_get_pdf_uri, uri, document_type
+    ):
+        """
+        GIVEN a document
+        WHEN a request is made with document uri
+        THEN html response should contain the download options div
+        AND this contains the Download PDF button
+        AND this contains the Download XML button
+        AND the descriptions refer to the document's type
+        """
+        mock_judgment.return_value = JudgmentFactory.build(uri=uri, is_published=True)
+        mock_get_pdf_size.return_value = "(112KB)"
+        mock_get_pdf_uri.return_value = "http://example.com/test.pdf"
+        client = Client()
+        response = client.get(f"/{uri}")
```

Comment:
Once routed into the view - the `judgement_uri` drops the starting slash. Is that an issue with a routing? It causes us to have to remove the / when creating our JudgmentFactory object

Reply:
So I think the canonical forms of the URL fragment are generally `ewhc/2009/1` and `/ewhc/2009/1.xml` depending on whether you're in Python or MarkLogic space, so I think this is correct. JudgmentFactory should probably be permissive about accepting `ewhc/2009/1`, though.

Reply:
interesting, yeah Ill try to edit the factory in a followup pr